workflow/cron/controller.go

```diff
@@ -141,17 +140,23 @@ func (cc *Controller) processNextCronItem() bool {
 		return true
 	}
 
-	cronWf, ok := obj.(*v1alpha1.CronWorkflow)
+	un, ok := obj.(*unstructured.Unstructured)
 	if !ok {
-		log.Warnf("Key '%s' in index is not a CronWorkflow", key)
+		log.WithField("key", key).Errorf("malformed cluster workflow template: expected *unstructured.Unstructured, got %s", reflect.TypeOf(obj).Name())
+		return true
+	}
+	cronWf := &v1alpha1.CronWorkflow{}
+	err = runtime.DefaultUnstructuredConverter.FromUnstructured(un.Object, cronWf)
```

Comment:
@jessesuen you asked how I dealt with malformed resources, in the case of cron we log the error and reject - this is what we already do